pkg/ottl/contexts/internal/logprofile/logging.go

```diff
@@ -13,6 +13,68 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+func getMapping(dict pprofile.ProfilesDictionary, idx int32) (mapping, error) {
+	mTable := dict.MappingTable()
+	if idx >= int32(mTable.Len()) {
+		return mapping{}, fmt.Errorf("mapping index out of bounds: %d", idx)
+	}
+	return newMapping(dict, mTable.At(int(idx)))
+}
+
+func getLocations(dict pprofile.ProfilesDictionary, start, length int32) (locations, error) {
+	locTable := dict.LocationTable()
+	if start >= int32(locTable.Len()) {
+		return locations{}, fmt.Errorf("location start index out of bounds: %d", start)
+	}
+	if start+length > int32(locTable.Len()) {
+		return locations{}, fmt.Errorf("location end index out of bounds: %d", start+length)
+	}
+
+	var joinedErr error
+	ls := make(locations, 0, length)
+	for i := range length {
+		l, err := newLocation(dict, locTable.At(int(start+i)))
```

Comment:
Location lookup has to get the table indices from `Profile.LocationIndices`.
This code is wrong in `main`, maybe a good chance to fix it here!?
Otherwise, let's open another PR to fix this. @TylerHelmuth

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/41510 fixes this separately.

Reply:
As https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/41510 adds another churn and more maintenance, I fixed the issue with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/41466/commits/6374fed46962bf7067376b6fa1145f00b19eac88.